src/main/java/org/alliancegenome/curation_api/dao/CrossReferenceDAO.java

```diff
@@ -1,14 +1,60 @@
 package org.alliancegenome.curation_api.dao;
 
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.persistence.Query;
+import jakarta.persistence.TypedQuery;
 import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
 import org.alliancegenome.curation_api.model.entities.CrossReference;
+import org.alliancegenome.curation_api.model.entities.Gene;
+import org.alliancegenome.curation_api.model.entities.ResourceDescriptorPage;
 
-import jakarta.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 @ApplicationScoped
 public class CrossReferenceDAO extends BaseSQLDAO<CrossReference> {
 
 	protected CrossReferenceDAO() {
 		super(CrossReference.class);
 	}
+
+	public Map<String, Long> getGenesWithCrossRefs(ResourceDescriptorPage page) {
+		String sql = """
+			select gc.genomicentity_id, cr.referencedcurie from genomicentity_crossreference as gc, crossreference as cr
+			where gc.crossreferences_id = cr.id AND cr.resourcedescriptorpage_id = :pageID
+			""";
+		Query query = entityManager.createNativeQuery(sql);
+		query.setParameter("pageID", page.getId());
+		List<Object[]> objects = query.getResultList();
+		Map<String, Long> ensemblGeneMap = new HashMap<>();
+		objects.forEach(object -> {
+			ensemblGeneMap.put((String) object[1], (Long) object[0]);
+		});
+		return ensemblGeneMap;
+	}
+
+	public Gene getGeneByCrossRef(String referencedCurie) {
+		String sql = """
+			select distinct gene from Gene gene, CrossReference as cc
+			where cc member of gene.crossReferences AND cc.referencedCurie = :refCurie
+			""";
```

Comment:
you should be able to do a: `select gene from Gene where gene.crossReferences.referencedCurie = :refCurie`

Reply:
That method is not being used at all. I am removing it.